### PR TITLE
CB-15373: Refreshing data mart DH does not work.

### PR DESCRIPTION
**Pull request by kkalvagadda1**
The current code was trying to update the external database configuration for Hive service type all the DH form factors.
This is not needed and is also causing issues for DH which does not have a Hive service type.

New code changes make sure that DH has service of HIVE before trying to update the database configuration.

### CODE REVIEW

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerConfigService.java [hunk 1]
```diff
@@ -110,6 +110,20 @@ private void modifyKnoxAutorestart(ApiClient client, String clusterName, boolean
...
+    public Optional<String> getServiceNameIfExists(ApiClient client, String clusterName, String serviceType) {
+        try {
+            ServicesResourceApi servicesResourceApi = clouderaManagerApiFactory.getServicesResourceApi(client);
+            ApiServiceList serviceList = servicesResourceApi.readServices(clusterName, DataView.SUMMARY.name());
+            return serviceList.getItems().stream()
+                    .filter(service -> serviceType.equalsIgnoreCase(service.getType()))
+                    .map(ApiService::getName)
+                    .findFirst();
+        } catch (ApiException e) {
+            LOGGER.debug("Failed to get {} service name from Cloudera Manager.", serviceType, e);
```
Comment (topolyai5) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerConfigService.java:122]: Could you please append the cluster name as well?

File: core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java [hunk 2]
```diff
@@ -95,25 +95,37 @@ public void accept(Event<ClusterStartRequest> event) {
...
-                        throw new CloudbreakException("Could not find RDS configuration for Hive");
+                if (apiConnectors.getConnector(stack).doesServiceExist(HIVE_SERVICE)) {
+                    //Update Hive service configuration
+                    Cluster cluster = clusterService.getById(datalakeStack.get().getCluster().getId());
+                    Optional<RDSConfig> rdsConfig = postgresConfigService.createRdsConfigIfNeeded(datalakeStack.get(), cluster, DatabaseType.HIVE)
+                            .stream().filter(config -> config.getType().toLowerCase().equals(DatabaseType.HIVE.toString().toLowerCase()))
+                            .findFirst();
+                    try {
+                        if (rdsConfig.isEmpty()) {
+                            throw new CloudbreakException("Could not find RDS configuration for Hive");
```
Comment (topolyai5) [inline core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java:121]: Maybe would be better if you switch the condition and just log the message if is empty. The stacktrace does not contain any useful information here. like
if (rdsConfig.present())
  //do the logic
else 
  //log the message

**Review by topolyai5 — CHANGES_REQUESTED** [review]

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerConfigService.java (hunk 1 above)
Comment (kkalvagadda1) [reply cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerConfigService.java:122]: Do you mean in the logs message?

File: core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java (hunk 2 above)
Comment (kkalvagadda1) [reply core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java]: Addressed your comment. Also made changes not to throw an exception in the failure case. Instead, I'm logging a failure. I don't want to throw an exception till we are sure that the failure that needs the flow to fail.

**Review by topolyai5 — APPROVED** [review]

**Review by horadla23 — CHANGES_REQUESTED** [review]
You are using a CM call to find out if a service is present in the cluster, but I think it is possible via database query from blueprint/template of the cluster. I would avoid CM call if possible, so pls at least check if it is possible to check service presence using template of the cluster.

File: core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java [hunk 3]
```diff
@@ -95,25 +95,38 @@ public void accept(Event<ClusterStartRequest> event) {
...
+                            .findFirst();
+                    try {
+                        if (rdsConfig.isPresent()) {
+                            apiConnectors.getConnector(stack).updateServiceConfig(HIVE_SERVICE, getRdsConfigMap(rdsConfig.get()));
+                        } else {
+                            LOGGER.error("Could not find RDS configuration for Hive");
+                        }
+                    } catch (CloudbreakException e) {
+                        LOGGER.info("Exception while updating the Data-Hub configuration", e);
                     }
```
[the remark this answers is not in the document]
Comment (kkalvagadda1) [reply core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java]: addressed

**Comment by kkalvagadda1** [comment]
> I would avoid CM call if possible, so pls at least check if it is possible to check service presence using template of the cluster.

let me check it.

**Review by horadla23 — APPROVED** [review]
LGTM

**Pull request merged**